{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. 다음 중 예외 처리에 대한 설명으로 잘못된 것을 모두 골라주세요\n",
    "\n",
    "<aside>\n",
    "🔑 정답 :  **예외는 현재 함수의 코드 블록에서만 처리할 수 있다.**\n",
    "\n",
    "           **finally는 예외가 발생하면 실행되지 않는다.**\n",
    "\n",
    "</aside>\n",
    "\n",
    "1. **예외는 현재 함수의 코드 블록에서만 처리할 수 있다. ⇒ 파이썬에서 예외는 발생한 위치에서부터 호출 스택을 거슬러 올라가면서 처리됩니다. 따라서 현재 함수 뿐만 아니라 해당 함수를 호출한 곳에서도 처리할 수 있습니다.**\n",
    "2. **finally는 예외가 발생하면 실행되지 않는다. ⇒ `finally` 블록은 예외 발생 유무에 상관없이 항상 실행됩니다. 예외가 발생하든 발생하지 않든, `finally` 블록의 코드는 항상 실행됩니다. 이는 자원 정리 등 예외 발생 여부와 상관 없이 반드시 실행해야 하는 코드를 작성하는 데 유용합니다.**\n",
    "\n",
    "---\n",
    "\n",
    "### 2. 예외를 임의로 발생 시키는 명령어를 골라주세요\n",
    "\n",
    "<aside>\n",
    "🔑 정답 :  **raise**\n",
    "\n",
    "</aside>\n",
    "\n",
    "예외를 임의로 발생시키는 명령어는 `raise`입니다. 파이썬에서 `raise`는 특정 예외를 강제로 발생시키는 데 사용됩니다. 이를 통해 예외 처리 흐름을 제어하거나, 예외 상황을 테스트하는 등의 작업을 수행할 수 있습니다.\n",
    "\n",
    "---\n",
    "\n",
    "### 3.  인덱스의 범위를 벗어났을 때 발생하는 에러에 대한 예외처리를 담당하는 코드를 골라주세요\n",
    "\n",
    "<aside>\n",
    "🔑 정답 : **IndexError**\n",
    "\n",
    "</aside>\n",
    "\n",
    "**인덱스의 범위를 벗어났을 때 발생하는 에러는 `IndexError`입니다.**\n",
    "\n",
    "파이썬에서 리스트나 튜플 등의 시퀀스 타입에서 인덱스 범위를 벗어나 접근하려고 할 때 이 에러가 발생합니다.\n",
    "\n",
    "---"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
